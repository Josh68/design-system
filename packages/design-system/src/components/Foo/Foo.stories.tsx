import type { Meta, StoryObj } from '@storybook/react';
import Foo from './Foo';
import { bool } from 'prop-types';

const meta: Meta = {
  component: Foo,
  argTypes: {
    children: { control: 'text' },
    bool: { control: 'boolean' },
  },
  args: {
    bool: false,
  },
  parameters: {
    docs: {
      underlyingHtmlElements: ['p'],
    },
  },
};

export default meta;

type Story = StoryObj<typeof Foo>;
const FooTemplate: Story = {
  render: ({ ...args }) => {
    return <Foo {...args}>{args.children}</Foo>;
  },
};

/**
 * This is a Foo component with some text.
 */

export const Default = {
  ...FooTemplate,
  args: {
    children: 'Hello, world!',
  },
};
