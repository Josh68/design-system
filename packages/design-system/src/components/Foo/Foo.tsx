import classNames from 'classnames';

/**
 * Foo does things!!!
 *
 * @param bool
 */
export const Foo = (props) => {
  const bgColor = props.bool ? 'mistyrose' : 'lightblue';
  return (
    <p
      style={{ backgroundColor: bgColor }}
      className={classNames('ds-u-color--base ds-u-padding--2', props.className)}
    >
      {props.children}
    </p>
  );
};

export default Foo;
