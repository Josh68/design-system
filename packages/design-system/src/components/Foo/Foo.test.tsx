import Foo from './Foo';
import { render, screen } from '@testing-library/react';

const defaultProps = {
  children: 'Hello, world!',
};

function renderFoo(props = {}) {
  return render(<Foo {...defaultProps} {...props} />);
}

describe('Foo', () => {
  it('should render a default foo', () => {
    renderFoo();
    const f = screen.getByText('Hello, world!');
    expect(f).toMatchSnapshot();
  });

  it('applies a mistyrose background', () => {
    renderFoo({ bool: true });
    const f = screen.getByText('Hello, world!');
    expect(f.style.backgroundColor).toContain('mistyrose');
  });
});
