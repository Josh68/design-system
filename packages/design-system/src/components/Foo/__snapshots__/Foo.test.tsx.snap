// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foo should render a default foo 1`] = `
<p
  class="ds-u-color--base ds-u-padding--2"
  style="background-color: lightblue;"
>
  Hello, world!
</p>
`;
